Assorted bugfixes made while debugging deployment to the COOL

jsf9k opened the pull request:
## 🗣 Description

This PR fixes a few bugs discovered while debugging deployment of OpenVPN to the COOL:
* Make adjustments for the fact that some modules now return resources, not ARNs or IDs.
* Stop using the subdomain input variable, and instead just specify the full hostname and the domain.
* Fix a broken PTR record.

## 💭 Motivation and Context

Without these changes, this module does not deploy correctly.

## 🧪 Testing

I deployed these changes to production and verified that the OpenVPN server was able to perform all its cloud-init tasks successfully and successfully connect to the domain (FreeIPA).

## 🚥 Types of Changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

jsf9k committed:
The cert read role module now returns resources, not ARNs or IDs

jsf9k committed:
Temporarily point the ssmreadrole module to an updated branch

jsf9k committed:
The ssm-read-role branch has been merged, so we can go back to using develop

jsf9k committed:
Cease using subdomain

Just specify the full hostname as the variable hostname and the domain
as the variable domain, like we do in
cisagov/freeipa-master-tf-module.

jsf9k committed:
Fix PTR record

jsf9k committed:
Shorten the PTR record name, since the rest appears to get filled in automagically

jsf9k commented on route53.tf:
```diff
@@ -32,23 +32,20 @@ resource "aws_route53_record" "server_AAAA" {
-    element(split(".", aws_instance.openvpn.private_ip), 2),
-    element(split(".", aws_instance.openvpn.private_ip), 1),
-    element(split(".", aws_instance.openvpn.private_ip), 0),
+    "%s",
```
@felddy, while fixing this I realized that Terraform and/or AWS appears to Do The Right Thing if you specify just enough of the name to "fill in" the PTR record.  For example, if this record were for the IP `10.11.12.13`, then you could provide the entire record name (`13.12.11.10.in-addr.arpa.`) or just the last octet (`13`).  If you do the latter, then look at the corresponding Route53 record in the AWS console, you can see that the `.12.11.10.in-addr.arpa.` part of the name has been automatically added.  With the previous code the record was coming out as `13.12.11.10.12.11.10.in-addr.arpa.`

I thought this might be useful information to point out to you, since you are our DNS expert.  At a minimum it allows us to create PTR records more succinctly.

felddy replied:
Wow.  I amazed this worked before and equally amazed this works with the last octet alone.  Thanks!
🤯

dav3r replied:
@jsf9k Your explanation in the comment above would make an excellent code comment to benefit future developers (and our own forgetful selves).

felddy reviewed (approved):
Looks good.  Thanks for brining this code to 2020!

dav3r reviewed (approved):
Outstanding! 🐞 🔨 
Please consider my one comment.

jsf9k committed:
Add comment describing interesting PTR record name behavior

@dav3r pointed out that this comment (which I shared in the pull
request) would be good to put in the code as well.

jsf9k committed:
Add missing period
